examples/model_selection/plot_learning_curve.py

```diff
@@ -1,218 +1,182 @@
+#
+# Here, we compute the learning curve of a naive Bayes classifier and a SVM
+# classifier with a RBF kernel using the digits dataset.
+from sklearn.datasets import load_digits
 from sklearn.naive_bayes import GaussianNB
 from sklearn.svm import SVC
-from sklearn.datasets import load_digits
+
+X, y = load_digits(return_X_y=True)
+naive_bayes = GaussianNB()
+svc = SVC(kernel="rbf", gamma=0.001)
+
+# %%
+# The :meth:`~sklearn.model_selection.LearningCurveDisplay.from_estimator`
+# displays the learning curve given the dataset and the predictive model to
+# analyze. To get an estimate of the scores uncertainty, this method uses
+# a cross-validation procedure.
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.model_selection import LearningCurveDisplay, ShuffleSplit
+
+fig, ax = plt.subplots(nrows=1, ncols=2, figsize=(10, 6), sharey=True)
+
+common_params = {
+    "X": X,
+    "y": y,
+    "train_sizes": np.linspace(0.1, 1.0, 5),
+    "cv": ShuffleSplit(n_splits=50, test_size=0.2, random_state=0),
+    "score_type": "both",
+    "n_jobs": 4,
+    "line_kw": {"marker": "o"},
+    "std_display_style": "fill_between",
+    "score_name": "Accuracy",
+}
+
+for ax_idx, estimator in enumerate([naive_bayes, svc]):
+    LearningCurveDisplay.from_estimator(estimator, **common_params, ax=ax[ax_idx])
+    handles, label = ax[ax_idx].get_legend_handles_labels()
+    ax[ax_idx].legend(handles[:2], ["Training Score", "Test Score"])
+    ax[ax_idx].set_title(f"Learning Curve for {estimator.__class__.__name__}")
+
+# %%
+# We first analyze the learning curve of the naive Bayes classifier. Its shape
+# can be found in more complex datasets very often: the training score is very
+# high when using few samples for training and decreases when increasing the
+# number of samples, whereas the test score is very low at the beginning and
+# then increases when adding samples. The training and test scores become more
+# realistic when all the samples are used for training.
+#
+# We see another typical learning curve for the SVM classifier with RBF kernel.
+# The training score remains high regardless of the size of the training set.
+# On the other hand, the test score increases with the size of the training
+# dataset. Indeed, it increases up to a point where it reaches a plateau.
+# Observing such a plateau is an indication that it might not be useful to
+# acquire new data to train the model since the generalization performance of
+# the model will not increase anymore.
+#
+# Complexity analysis
+# ===================
+#
+# In addition to these learning curves, it is also possible to look at the
+# scalability of the predictive models in terms of training and scoring times.
+#
+# The :class:`~sklearn.model_selection.LearningCurveDisplay` class does not
+# provide such information. We need to resort to the
+# :func:`~sklearn.model_selection.learning_curve` function instead and make
+# the plot manually.
```

Comment:
Don't you think it would be worth to enable this kind of plot ? I feel like it doesn't add too much complexity and I can clearly see the value

Reply:
I agree. I would postpone it for another PR. It will be easier to review with fewer changes. Once this PR merged, I will open a subsequent issue to not forget about this improvement.